cpp/log/cert.cc

```diff
@@ -38,6 +38,35 @@ static int X509_get_signature_nid(const X509* x) {
 
 namespace cert_trans {
 
+// Useful because util::Status clashes with Cert::Status so we have to
+// refer to it within the namespace while Cert::Status exists.
+static const util::Status& ERROR_STATUS = util::Status::UNKNOWN;
+
+// TODO. These helpers can be removed when Cert::Status has been removed.
+// They make it easier to do an incremental removal
+
+static Cert::Status StatusOrBoolToCertStatus(StatusOr<bool> status) {
```

Comment:
Well I'm having to work without the tools I'd normally have available to
automate some of this but I did do an analysis before I started.

Because some of the logic is complex and the danger of returning a
StatusOr<bool> instead of int or vice-versa I used the helpers to make the
intermediate state read more like what was there currently.


On 28 August 2015 at 16:59, Pierre Phaneuf <notifications@github.com> wrote:

> In cpp/log/cert.cc
> <https://github.com/google/certificate-transparency/pull/885#discussion_r38214880>
> :
>
> > @@ -38,6 +38,35 @@ static int X509_get_signature_nid(const X509* x) {
> >
> >  namespace cert_trans {
> >
> > +// Useful because util::Status clashes with Cert::Status so we have to
> > +// refer to it within the namespace while Cert::Status exists.
> > +static const util::Status& ERROR_STATUS = util::Status::UNKNOWN;
> > +
> > +// TODO. These helpers can be removed when Cert::Status has been removed.
> > +// They make it easier to do an incremental removal
> > +
> > +static Cert::Status StatusOrBoolToCertStatus(StatusOr<bool> status) {
>
> Haha!
>
> If you start from a function that doesn't return a Cert::Status,
> StatusOrBoolToCertStatus wouldn't have needed to exist, and if you start
> from a function that doesn't call anything returning a Cert::Status, you
> wouldn't have needed CertStatusToStatusOrBool (but that can be harder,
> for example in #892
> <https://github.com/google/certificate-transparency/pull/892>, where I'm
> having to carry around GetVerifyError a bit longer, but fewer and fewer
> code paths are needing it).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/885/files#r38214880>
> .
>